app/deployments/routes.py

```diff
@@ -406,7 +413,9 @@ def get_vm_info(depid):
     def find_node_with_pubip(resources):
         for resource in resources:
             if "VirtualMachineInfo" in resource["metadata"]:
-                vm_info = json.loads(resource["metadata"]["VirtualMachineInfo"])["vmProperties"]
+                vm_info = json.loads(resource["metadata"]["VirtualMachineInfo"])[
+                    "vmProperties"
```

Comment:
You're checking that `VirtualMachineInfo` exists but not checking if `vmProperties` exists in that JSON.
Could be ok if it is an always present property, but still dangerous in case of missing key.
Are eventual errors handled?